packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -866,14 +880,75 @@ export default function makeKernelKeeper(kernelStorage, kernelSlog) {
     kvStore.set(`${kernelSlot}.data.slots`, capdata.slots.join(','));
   }
 
-  function cleanupAfterTerminatedVat(vatID) {
+  function removeVat(vatID) {
+    // Delete primary swingstore records for this vat, in preparation
+    // for (slow) deletion. After this, swingstore exports will omit
+    // this vat. This is called from the kernel's terminateVat, which
+    // initiates (but does not complete) deletion.
+    snapStore.stopUsingLastSnapshot(vatID);
+    transcriptStore.stopUsingTranscript(vatID);
+  }
+
+  /**
+   * Perform some cleanup work for a specific (terminated but not
+   * fully-deleted) vat, possibly limited by a budget. Returns 'done'
+   * (where false means "please call me again", and true means "you
+   * can delete the vatID now"), and a count of how much work was done
+   * (so the runPolicy can decide when to stop).
+   *
+   * @param {string} vatID
+   * @param {number} [budget]
+   * @returns {{ done: boolean, cleanups: number }}
+   *
+   */
+  function cleanupAfterTerminatedVat(vatID, budget = undefined) {
     insistVatID(vatID);
+    let cleanups = 0;
+    const work = {
+      exports: 0,
+      imports: 0,
+      kv: 0,
+      snapshots: 0,
+      transcripts: 0,
+    };
+    let spend = _did => false; // returns "stop now"
+    if (budget !== undefined) {
+      assert.typeof(budget, 'number');
+      spend = (did = 1) => {
+        assert(budget !== undefined); // hush TSC
+        cleanups += did;
+        budget -= did;
+        return budget <= 0;
+      };
+    }
```

Comment:
We can remove some code and eliminate some boolean traps:
```suggestion
  function cleanupAfterTerminatedVat(vatID, budget = Infinity) {
    insistVatID(vatID);
    let cleanups = 0;
    const work = {
      exports: 0,
      imports: 0,
      kv: 0,
      snapshots: 0,
      transcripts: 0,
    };
```
and replace every
```js
      if (spend()) {
        logWork();
        return { done: false, cleanups };
      }
```
with
```js
      cleanups += 1;
      if (cleanups >= budget) {
        logWork();
        return { done: false, cleanups };
      }
```
and `vatKeeper.deleteFoo(budget)` with `vatKeeper.deleteFoo(budget - cleanups)`.

Reply:
I like the cleanup (and thanks for the reminder that `Infinity` exists), but I think I'm going to defer this one until I figure out how to let the policy set separate budgets for the different phases: the `budget` argument might want to be a record that defaults to `{ exports: Infinity, imports: Infinity, etc }`, in which case I might need to bring back `spend(phase, count = 1)`.